### ISSUE TITLE
Could not save file: Permission denied

### ISSUE THREAD

**Issue by terba** [gphalkes/tilde#18]
Files _with_ write permission in a directory _without_ write permission can not be saved.

Example:
```bash
mkdir test
touch test/test.txt
chmod 500 test
tilde test/test.txt #You can not save the modifications!
vi test/test.txt #Can save
mcedit test/test.txt #Can save
```

**Comment by gphalkes** [comment]
The reason for this is that Tilde doesn't overwrite the file itself, but rather writes a new file and renames it over the old file. Tilde does it this way to prevent the situation in which the file is only partially written and then interrupted, but it can indeed lead to this problem.

That is not to say that the current situation is right. There should be a fallback for the case where the directory is unwritable, where it should overwrite the existing file directly. Leaving the bug open until this is implemented.

**Comment by gphalkes** [comment]
Fixed in https://github.com/gphalkes/tilde/commit/5aae9aa0754c0deff54406c9f1a9c9a85614c8b5, and will be part of the next release.

**Comment by gphalkes** [comment]
This is fixed in Tilde 0.4.0
